.github/workflows/doc-tests.yml

```
@@ -7,30 +7,23 @@ name: Test documentation
 ## The tests only need to run on one OS -- Linux is used because the test scripts are bash
 
 on:
-  push:
-    branches: [ master ]
+  workflow_dispatch:
   pull_request:
     branches: [ master ]
+  push:
 
 jobs:
-  build:
+  check-deep-tests:
+     uses: ./.github/workflows/check-deep-tests-reusable.yml
+     with:
+       branch: master
 
-    runs-on: ${{ matrix.os }}
-    strategy:
-      fail-fast: false
-      matrix:
-        include:
-          - os: ubuntu-latest
-            z3: z3-4.8.5-x64-ubuntu-16.04
-            chmod: true
+  doctests:
+    needs: check-deep-tests
+    if: always() && ((needs.check-deep-tests.result == 'success' || contains(github.event.pull_request.labels.*.name, 'run-deep-tests') && ( github.event_hame != 'push' || github.ref_name == 'master' || vars.TEST_ON_FORK == 'true' ))) 
```

```suggestion
    if: always() && ((needs.check-deep-tests.result == 'success' || contains(github.event.pull_request.labels.*.name, 'run-deep-tests')) && (github.event_name != 'push' || github.ref_name == 'master' || vars.TEST_ON_FORK == 'true'))
```
I think this could work, but ideally, I'd like us to try it on a fork first. I'm not sure about precedence here, but the typo also appears in other places.